packages/react-server-dom-webpack/src/ReactFlightClientConfigNodeBundler.js

```diff
@@ -71,14 +77,21 @@ export function resolveServerReference<T>(
   const idx = id.lastIndexOf('#');
   const specifier = id.substr(0, idx);
   const name = id.substr(idx + 1);
-  return {specifier, name};
+  return {specifier, name, ssrChunks: null};
 }
 
 const asyncModuleCache: Map<string, Thenable<any>> = new Map();
 
 export function preloadModule<T>(
   metadata: ClientReference<T>,
 ): null | Thenable<any> {
+  const ssrChunks = metadata.ssrChunks;
+  if (ssrChunks) {
+    for (let i = 0; i < ssrChunks.length; i++) {
+      const href = ssrChunks[i];
+      preloadModuleForSSR(href);
```

Comment:
As I'm reading through this pull request, I can see that this is intended for Float to render `<link rel="preload" as="script" href="/some-client-chunk.js" />`. As someone who is not as deeply involved in the project, I thought it might be helpful to share an outsider's perspective. From my point of view, renaming `ssrChunks` to `clientChunks` and `preloadModuleForSSR` to `preloadModuleForHydration` could potentially make the code more understandable without context.

Reply:
This is the general intent of the code. I’m currently reworking the implementation significantly so none of these names are likely to stick around